ui/pages/asset/components/asset-chart.tsx

```diff
@@ -0,0 +1,307 @@
+import React, { useEffect, MouseEvent, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  Chart,
+  LineElement,
+  LinearScale,
+  PointElement,
+  Filler,
+  ChartData,
+  ChartOptions,
+  Decimation,
+  Point,
+  ChartEvent,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+import classnames from 'classnames';
+import { getCurrentChainId, getCurrentCurrency } from '../../../selectors';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  Display,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+  FlexDirection,
+} from '../../../helpers/constants/design-system';
+import {
+  Box,
+  ButtonSecondary,
+  ButtonSecondarySize,
+  Icon,
+  IconName,
+  IconSize,
+  Text,
+} from '../../../components/component-library';
+import { hexToDecimal } from '../../../../shared/modules/conversion.utils';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import fetchWithCache from '../../../../shared/lib/fetch-with-cache';
+import { MINUTE } from '../../../../shared/constants/time';
+import AssetPrice from './asset-price';
+import ChartTooltip from './chart-tooltip';
+
+/** Time range units supported by the price API */
+type TimeRange = `${number}D` | `${number}M` | `${number}Y`;
+
+/** A Chart.js plugin that draws a vertical crosshair on hover */
+const crosshairPlugin = {
+  id: 'crosshair',
+  afterEvent(chart: Chart, { event }: { event: ChartEvent }) {
+    chart.crosshairX = event.type === 'mouseout' ? undefined : event.x;
+    chart.draw();
+  },
+  afterDraw(chart: Chart) {
+    if (chart.crosshairX !== undefined) {
+      const { y } = chart.scales;
+      chart.ctx.beginPath();
+      chart.ctx.moveTo(chart.crosshairX, y.getPixelForValue(y.max));
+      chart.ctx.lineWidth = 2;
+      chart.ctx.strokeStyle = '#BBC0C5';
+      chart.ctx.lineTo(chart.crosshairX, y.getPixelForValue(y.min));
+      chart.ctx.stroke();
+    }
+  },
+};
+
+Chart.register(
+  LinearScale,
+  PointElement,
+  LineElement,
+  Filler,
+  Decimation,
+  crosshairPlugin,
+);
+
+const initialChartOptions: ChartOptions<'line'> = {
+  normalized: true,
+  parsing: false,
+  aspectRatio: 2.3,
+  layout: { autoPadding: false, padding: 0 },
+  animation: { duration: 0 },
+  fill: true,
+  borderColor: '#0376c9',
+  backgroundColor: ({ chart }: { chart: Chart }) => {
+    const gradient = chart.ctx.createLinearGradient(0, 0, 0, chart.height);
+    gradient.addColorStop(0, '#0376c94C');
+    gradient.addColorStop(1, '#D9D9D900');
```

Comment:
If CSS variables don't work we could import the `brandColor` object for now. It might be beneficial to implement a theme provider that exposes the JS tokens for the hex values throughout the extension. The `brandColor` object is in v2 of the design tokens but I think the CSS exports are currently broken so might have to let this one slide and update it post merge. cc @garrettbear any ideas?

```
import { brandColor } from '@metamask/design-tokens';
```

Reply:
In https://github.com/MetaMask/metamask-extension/pull/22508/commits/a6260a1f09a2d7a081da8c715f522fd4f3024d51 I changed it to use `brandColor.blue400` or `brandColor.blue500` depending on dark/light theme